app/src/main/java/org/fossasia/pslab/activity/OscilloscopeActivity.java

```diff
@@ -740,6 +757,44 @@ public void setXAxisLabel(String xAxisInput) {
         xAxisLabel.setText(xAxisInput);
     }
 
+    public void howToUseDialog(String title, String intro, int iconID, String desc) {
+
+        final AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        LayoutInflater layoutInflater = getLayoutInflater();
+        View dialogView = layoutInflater.inflate(R.layout.custom_dialog_box, null);
+        builder.setView(dialogView);
+        builder.setTitle(title);
+
+        final TextView oscGuideText = dialogView.findViewById(R.id.custom_dialog_text);
+        final ImageView oscGuideImage = dialogView.findViewById(R.id.custom_dialog_schematic);
+        final TextView oscDescription = dialogView.findViewById(R.id.description_text);
+        final CheckBox doNotShowDialog = dialogView.findViewById(R.id.toggle_show_again);
+        final Button dismissButton = dialogView.findViewById(R.id.dismiss_button);
+
+        oscGuideText.setText(intro);
+        if (iconID != 0)
+            oscGuideImage.setImageResource(iconID);
+        oscDescription.setText(desc);
+
+        final SharedPreferences sharedPreferences = getSharedPreferences(TAG+title, 0);
```

Comment:
Let's use a common sharedpreference across every class just like the name implies `shared` :) And use private mode instead of `0`